.github/workflows/mac_m1_binary.yml

```diff
@@ -1,14 +1,29 @@
-name: MacOS-m1 binary release
+name: Mac OS binary release
```

Comment:
I guess a follow up question: who uses these binaries at all? It isn't like macos is a valid production target, its only supported for development. So if we aren't building code every commit and the binary artifacts oughtn't be used for prod, why spend time building any binaries at all?